src/toil/job.py

```diff
@@ -477,6 +477,27 @@ def checkNewCheckpointsAreLeafVertices(self):
                 if len(y._children) != 0 and len(y._followOns) != 0 and len(y._services) != 0:
                     raise JobGraphDeadlockException("New checkpoint job %s is not a leaf in the job graph" % y)
 
+    def defer(self, func, *args, **kwargs):
+        """
+        This function is used to register a function that will be run after the completion of the
```

Comment:
It's very important that this is precise. Consider this as the first paragraph:

> Register a deferred function, i.e. a callable that will be invoked after the current attempt at running this job concludes. A job attempt is said to *conclude* when the job function (or the `Job.run` method for class-based jobs) returns, raises an exception or after the process running it terminates abnormally. A deferred function will be called on the node that attempted to run the job, even if a subsequent attempt is made on another node. A deferred function should be idempotent because it may be called multiple times on the same node or even in the same process. More than one deferred function may be registered per job attempt by calling this method repeatedly with different arguments. If the same callable is registered twice, it will be called twice per job attempt.